tests/unit/dataactcore/test_job_queue.py

```diff
@@ -0,0 +1,46 @@
+from dataactcore.utils.jobQueue import JobQueue
+from dataactcore.config import CONFIG_BROKER
+from unittest.mock import Mock
+from pytest import raises
+from dataactcore.utils.responseException import ResponseException
+import os.path
+
+
+def test_generate_d_file_success(monkeypatch):
+    """ Test successful generation of D1 and D2 files """
+    local_file_name = "12345_test_file.csv"
+
+    file_path = "".join([CONFIG_BROKER['d_file_storage_path'], local_file_name])
+
+    with open(file_path, "w") as file:
+        file.write("test")
+
+    result_xml = "<results>test_file.csv</results>"
+
+    jq = JobQueue()
+    monkeypatch.setattr(jq, 'update_d_file_status', Mock())
+    monkeypatch.setattr(jq, 'download_file', Mock())
+    monkeypatch.setattr(jq, 'get_xml_response_content', Mock(return_value=result_xml))
+
+    jq.generate_d_file(Mock(), 1, 1, Mock(), local_file_name, True)
+
+    assert os.path.isfile(file_path)
```

Comment:
Minor, but one testing style that I think is particularly good:
1. assert stable state
2. run function
3. assert changed state

Here that'd look like:
```python
assert not os.path.isfile(file_path)
jq.generate_d_file(Mock(), 1, 1, Mock(), local_file_name, True)
assert os.path.isfile(file_path)
```

This not only shows the reader what's changing, but also avoids false negatives